pages/index.js

```diff
@@ -18,36 +21,129 @@ import {
   LatestBlockTXNIcon,
   SecondsToBlockIcon,
   TotalSupplyIcon,
-} from 'svgx'
+} from 'svgx';
 
+const LastBlockInfo = () => {
+  const cardWidth = useBreakpointValue({
+    base: '100%',
+    sm: 'calc(50% - 1rem)',
+    md: 'calc(33% - 1rem)',
+  });
+  const headBlock = useBlockHead();
 
-const useBlocks = () => ({
-  loaded: true,
-  data: new Array(14).fill({
-    sequence: 23940194,
-    size: 4096,
-    transactions_count: 1,
-    hash: '0000456456654564346345644564000045645665456434634564456445644564',
-    timestamp: '6/22/2021 8:09:45PM'
-  })
-})
+  const getValue = (field, transform = (value) => value) => {
+    return headBlock.loaded ? (
+      transform(headBlock.data[field])
+    ) : (
+      <span>&nbsp;</span>
+    );
+  };
+
+  return (
+    <Flex w="100%" wrap="wrap" mb="2.25rem">
+      <Card
+        m="0.5rem"
+        w={cardWidth}
+        label="Difficulty"
+        value={getValue('difficulty')}
+        icon={<DifficultyIcon />}
+        isLoading={!headBlock.loaded}
+      />
+      <Card
+        m="0.5rem"
+        w={cardWidth}
+        label="Height"
+        value={getValue('sequence')}
+        icon={<HeightIcon />}
+        isLoading={!headBlock.loaded}
+      />
+      <Card
+        m="0.5rem"
+        w={cardWidth}
+        label="Latest block hash"
+        value={getValue('hash', truncateHash)}
+        icon={<LatestBlockHashIcon />}
+        isLoading={!headBlock.loaded}
+      />
+      <Card
+        m="0.5rem"
+        w={cardWidth}
+        label="Latest Block txn"
+        value={getValue('transactions_count')}
+        icon={<LatestBlockTXNIcon />}
+        isLoading={!headBlock.loaded}
+      />
+      <Card
+        m="0.5rem"
+        w={cardWidth}
+        label="Seconds to block"
+        value={getValue('time_since_last_block_ms', (value) =>
+          Math.floor(value / 1000),
+        )}
+        icon={<SecondsToBlockIcon />}
+        isLoading={!headBlock.loaded}
+      />
+      <Card
+        m="0.5rem"
+        w={cardWidth}
+        label="Total Supply"
+        value="-"
+        icon={<TotalSupplyIcon />}
+        isLoading={!headBlock.loaded}
+      />
+    </Flex>
+  );
+};
+
+const LatestBlocks = () => {
+  const { loaded, data, error } = useBlocks({
+    limit: 10,
+    main: true,
+  });
+
+  return (
+    <Flex direction="column" mb="2rem">
+      <Text fontSize="1.5rem" mb="0.625rem">
+        Latest Blocks
+      </Text>
+      {!error ? (
+        <BlocksTable
+          data={loaded ? data.data : Array.from({ length: 10 }, () => null)}
```

Comment:
length should have the same value as limit in `useBlocks` hook so it make sense to pass it as props or make a constant

Reply:
Might make more sense idiomatically to use `Array.fill(null)` instead of `() => null`?